gems/pending/spec/spec_helper.rb

```diff
@@ -32,3 +36,30 @@
   config.backtrace_exclusion_patterns << %r{/lib\d*/ruby/[0-9]}
   config.backtrace_exclusion_patterns << %r{/gems/[0-9][^/]+/gems/}
 end
+
+#
+# So tests can clear class-level caches between examples.
+# TODO: Add this to the azure-armrest gem.
+#
+class Azure::Armrest::ArmrestService
+  def self.clear_caches
+    @@providers_hash = {}
+    @@tokens         = {}
+    @@subscriptions  = {}
+  end
+end
```

Comment:
@chrisarcand - The immediate need for this is for the sake of the specs. I was going to go the `class_eval` route initially, but this seemed cleaner. I like the _test macos_ route, but then the question becomes, where should the macros live?

To @jrafanie 's point, all of the ruby files under support are loaded by the `spec_helper.rb` anyway, so we'd still be requiring azure-armrest for all the specs. Maybe we should have a `spec/macros` directory. The files under which would only be loaded by the specs that need them.